model/postgres/postgres_server.rb

```diff
@@ -315,8 +315,8 @@ def switch_to_new_timeline(parent_id: timeline.id)
       timeline_id: Prog::Postgres::PostgresTimelineNexus.assemble(location_id: resource.location_id, parent_id: parent_id).id,
       timeline_access: "push"
     )
-
-    refresh_walg_credentials
+    incr_configure_s3_new_timeline if timeline.aws?
```

Comment:
The point of separating the `aws`/`metal` implementations into separate files is to avoid manual `aws?` checks in the main model files. The simplest way to handle this case is to add `{aws,metal}_incr_s3_new_timeline`  methods to the appropriate files, have the `aws` implementation call `incr_configure_s3_new_timeline`, have the `metal` implementation do nothing, and call `incr_s3_new_timeline` here.

Reply:
Well, you can choose a different method name. The only requirement is the method name cannot be the same as `incr_configure_s3_new_timeline`.